warehouse.tf

```diff
@@ -44,17 +45,22 @@ data "aws_iam_policy_document" "redshift_load_policy" {
   }
 }
 
-resource "aws_security_group" "allow_access_to_redshift_from_vpn" {
-  name        = "Allow Prod VPN access to Redshift"
-  description = "Allow inbound access from ControlShift VPN"
+resource "aws_redshift_subnet_group" "default" {
+  name = "subnet-group-for-redshift-cluster"
+  subnet_ids = [ aws_subnet.private_redshift_subnet_shared_with_lambdas_and_glue.id ]
+}
+
+resource "aws_security_group" "allow_access_to_redshift_from_glue" {
+  name        = "Allow AWS Glue access to Redshift"
+  description = "Allow inbound access from all servers within the security group, and allow full outbound access"
   vpc_id      = data.aws_vpc.default.id
 
   ingress {
```

Comment:
What is this used for exactly, allowing external access to our test example redshift? Can we replace with a variable to force people using this example plan to make a choice about who has access?

Reply:
compared to the description this seems to allow access from the entire internet, not just glue subnet.

Reply:
The `self = true` on the route means it only allows traffic from the same security group, so this isn't limiting by IP address but by internal traffic (see docs for `ingress` block at https://www.terraform.io/docs/providers/aws/r/security_group.html)

Reply:
And here's Glue docs with the requirement to allow full access within the security group: https://docs.aws.amazon.com/glue/latest/dg/setup-vpc-for-glue-access.html